Civi/Api4/Action/GetActions.php

```diff
@@ -79,7 +79,8 @@ private function loadAction($actionName, $method = NULL) {
     try {
       if (!isset($this->_actions[$actionName]) && (!$this->_actionsToGet || in_array($actionName, $this->_actionsToGet))) {
         $action = \Civi\API\Request::create($this->getEntityName(), $actionName, ['version' => 4]);
-        if (is_object($action) && (!$this->checkPermissions || $action->isAuthorized(\CRM_Core_Session::singleton()->getLoggedInContactID()))) {
+        $authorized = \Civi::service('civi_api_kernel')->runAuthorize($this->getEntityName(), $actionName, ['version' => 4]);
+        if (is_object($action) && (!$this->checkPermissions || $authorized)) {
```

Comment:
Whoops @seamuslee001 I merged too soon -- this is hurting performance by calling  `runAuthorize` even when `$this->checkPermissions == FALSE`